Sources/Kitura/SwaggerGenerator.swift

```diff
@@ -1076,6 +1075,42 @@ public struct SwaggerDocument: Encodable {
         }
         return document
     }
+    
+    private static func getTypeName(type: Any.Type) -> String {
+        let fullName = String(reflecting: type)
+        return removeFrameworkname(fullName: fullName)
+    }
+    
+    private static func getTypeName(type: Any) -> String {
+        let fullName = String(reflecting: type)
+        return removeFrameworkname(fullName: fullName)
+    }
+    
+    private static func getTypeName(typeInfo: Any.Type) -> String {
+        let fullName = String(reflecting: typeInfo)
+        return getTypeName(fullName: fullName)
+    }
+    
+    private static func getTypeName(typeInfo: TypeInfo) -> String {
+        let fullName = String(reflecting: typeInfo)
+        return getTypeName(fullName: fullName)
+    }
+    
+    private static func getTypeName(fullName: String) -> String {
+        let fullName = fullName
+            .replacingOccurrences(of: ".keyed: ", with: "")
+            .replacingOccurrences(of: ".single: ", with: "")
```

Comment:
I kind of see what's going on here, but I'm not convinced that converting a TypeInfo to a String and then removing the extra text is a good strategy.  Also this ignores other cases (SwaggerGenerator doesn't properly handle dictionaries and arrays nested within each other currently, but we might find `.cyclic` and `.optional` types).

Instead, you could switch over the `TypeInfo` in `getTypeName(typeInfo:)` and extract the value you need.  You could use TypeInfo's [debugDescription](https://github.com/IBM-Swift/TypeDecoder/blob/master/Sources/TypeDecoder/TypeDecoder.swift#L244) as a reference.

Reply:
Cleaned up the code